src/JSInterop/Microsoft.JSInterop/src/IJSRuntime.cs

```diff
@@ -24,6 +25,21 @@ public interface IJSRuntime
     /// <returns>An instance of <typeparamref name="TValue"/> obtained by JSON-deserializing the return value.</returns>
     ValueTask<TValue> InvokeAsync<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string identifier, object?[]? args);
 
+    /// <summary>
+    /// Invokes the specified JavaScript function asynchronously.
+    /// <para>
+    /// <see cref="JSRuntime"/> will apply timeouts to this operation based on the value configured in <see cref="JSRuntime.DefaultAsyncTimeout"/>. To dispatch a call with a different timeout, or no timeout,
+    /// consider using <see cref="InvokeAsync{TValue}(string, CancellationToken, object[])" />.
+    /// </para>
+    /// </summary>
+    /// <typeparam name="TValue">The JSON-serializable return type.</typeparam>
+    /// <param name="identifier">An identifier for the function to invoke. For example, the value <c>"someScope.someFunction"</c> will invoke the function <c>window.someScope.someFunction</c>.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> to use for JSON serialization and deserialization.</param>
+    /// <param name="args">JSON-serializable arguments.</param>
+    /// <returns>An instance of <typeparamref name="TValue"/> obtained by JSON-deserializing the return value.</returns>
+    ValueTask<TValue> InvokeAsync<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string identifier, JsonSerializerOptions options, object?[]? args)
+        => throw new InvalidOperationException($"Supplying a custom {nameof(JsonSerializerOptions)} is not supported by the current JS runtime");
+
```

Comment:
It was my understanding that we don't want to do this at all.

Is this required to improve the perf startup? How much improvement do we lose if we don't have it?

Adding support for passing custom JsonSerializer options to JSInvoke method calls broadens the scope of this feature quite a lot and opens us up to a lot of cases where things can go wrong in subtle and hard to debug ways.

For that reason, I would prefer if we stuck to making whatever functionality we need internal and maintain our current policy that this part of the system is not configurable, and that if you want to perform custom serialization you can perform the serialization yourself and pass in a byte array.

Reply:
My interpretation was that adding this to `IJSRuntime` is necessary for the layering to continue to make sense. I was assuming it's not possible to restrict this to `WebAssemblyJSRuntime` since the relevant calls come from other layers, but if that's incorrect and it can in fact be restricted that way it's worth considering.

> I also don't want us to live in a world where this is supported in some platforms and not in others. Every time we introduce something like that, its a headache for customers.

This wouldn't be an issue in practice. This PR adds the support to the underlying `JSRuntime` base class which we inherit in all the supported platforms. It would impact any 3rd-party `IJSRuntime` implementations, but that's realistically only in tests. 

* In unit tests, the new calls would not be exercised unless people start using the new APIs, so it's not an issue.
* In integration tests like bUnit, I'm unsure whether the new calls would be exercised, but I think bUnit would want to add support for this anyway